### PR TITLE
Add Ethernet support for NXP S32Z/E SoCs

### PR DESCRIPTION
**Pull request by manuargue**
This PR introduces support for Ethernet on NXP S32Z/E SoCs, specifically for S32Z27x-DC(2) board.

The Ethernet low-level driver is based on NXP S32 Network Controller (NETC) complex, which is composed of an External MDIO controller, partitionable Station Interfaces (SI) and an integrated Ethernet Switch. Current driver allows to manage from Zephyr a Physical Station Interface (PSI) and/or a Virtual SI (VSI). The rest of the VSI's shall be assigned to different cores of the system. The Switch is initialized from the Ethernet shim driver with a default configuration, and all ports are enabled and are transparent for the user. A separate Switch driver will be addressed in a future PR as needed.

NETC uses MSI-X, therefore a new Mbox-based driver is implemented for the Message Receive Unit (MRU) peripheral, to be able to terminate in interrupts targetting the core controlling the SI.

### CODE REVIEW

**Comment by manuargue** [comment]
Hello @tbursztyka, @rlubos, @carlocaione, could you please provide some feedback on these changes? Thanks

**Review by dleach02 — DISMISSED** [review]

**Comment by dleach02** [comment]
@manuargue the HAL PR was merged a couple weeks back so I would update the west.yml (or maybe rebase if the HAL changes are already part of main)

**Comment by manuargue** [comment]
> @manuargue the HAL PR was merged a couple weeks back so I would update the west.yml (or maybe rebase if the HAL changes are already part of main)

already did that in last rebase/push, but for some reason it didn't remove the DNM label on this pr... re-pushed

**Comment by manuargue** [comment]
> @manuargue the HAL PR was merged a couple weeks back so I would update the west.yml (or maybe rebase if the HAL changes are already part of main)

already rebased and removed changes in manifest on previous update since hal_nxp is already up to date, but for some reason the DNM label is not removed. Rebased again.

**Review by dleach02 — APPROVED** [review]

**Review by EmilioCBen — APPROVED** [review]

File: drivers/mbox/Kconfig
Comment (danieldegrasse) [inline drivers/mbox/Kconfig:14]: Minor comment (non-blocking): Is there an MBOX driver that needs to override `CONFIG_MBOX_INIT_PRIORITY`? I assume that this pattern was followed due to the way that the "root" MDIO drivers KConfig file appears to be written, but in my experience the individual driver KConfig files are typically sourced after the primary KConfig file has defined all symbols

**Review by danieldegrasse — APPROVED** [review]

**Comment by laurenmurphyx64** [comment]
Ping @tbursztyka. Sorry if this is annoying. I can't merge without the assignee's approval and I thought it'd be helpful to get this back on your docket before feature freeze this Friday.

**Pull request merged**
